cpp/include/kvikio/shim/cufile_h_wrapper.hpp

```diff
@@ -75,7 +75,7 @@ CUfileError_t cuFileDriverSetMaxPinnedMemSize(...);
 
 #endif
```

Comment:
It seems like this would be safer if we nested this macro so that we got
```
#ifdef KVIKIO_CUFILE_FOUND
...
  #ifndef KVIKIO_CUFILE_BATCH_API_FOUND
  ...
  #endif
#endif
```

Currently it seems like we are defining these other batch/stream types even if `KVIKIO_CUFILE_FOUND` is false? Or I'm misreading the macros... basically I am proposing that we move the `#endif` on line 76 to the bottom of the file.